src/lib/controllers/app-controller.js

```diff
@@ -14,17 +14,37 @@ import EnUsMainMenu from '@/locales/en-us/messages-main-menu.json'
 import enGB from '@/locales/en-gb/messages.json'
 
 export default class AppController {
+  /**
+   *
+   * @param {String} appId - id attribute of the HTML element where Vue application should be attached
+   */
   constructor ({ appId }) {
-    this.attachVueComponents(appId)
+    if (!appId) {
+      console.error('You should define id inside AppController initialization to start the application.')
+      return

```

Comment:
It was exactly my aim to have such a workflow if `appId` is not defined.

We have this way of initialization of app in `index.html` :
```
        document.addEventListener("DOMContentLoaded", function(event) {
          new window.AlignmentEditor.AppController({
            appId: 'alpheios-alignment-editor'
          }).init()
        });
```
it works the following way:
* it creates AppController 
* checks if it has `appId` (later we could have here more parameters)
* prints an error that we don't have `appId`
* executes `init`

Inside `init` we have no actions if `appId` is not defined, so the application won't be created and attached to HTML layout.
Also we have only one printed error in console. But in future in some cases we could do actions even without creating visual parts of application.

I believe that this way we have more flexible approach - we could use app in different states.
Also we don't need to have try/catch block in HTML definition, and this way we have easier way to embed application for anyone. 

@balmas and @kirlat , do you agree with this approach?

Reply:
On my opinion, if the AppController is planned to be used without the UI then having no `appId` is normal and we should not break the constructor flow because of that. We could just log a warning that the UI will not be available because no `appId` is provided.

In the `init()` we can check for the presence of the `appId` as we do now and do nothing if it's not there. Or, maybe better yet, we should have something like `isUIAvailable` flag that will be set to `false` in the constructor if not enough information to create the UI is provided. Why I like the flag approach better is because it, on my opinion, makes code more expressive. The name of the flag could clearly reflect what it is about while from the `appId` name it is not so clear.

What do you think?

Reply:
I think that here we have two steps of initializing App on the page:
 - AppController is created with some parameters (and we could define here some additional actions and checks with parameters)
- AppController starts to work based on the given parameters - here would be actions (now it is create Vue and attach to the page)

If to be honest from one side I like flag approach - it makes code more abstract, but if it is used only once - it looks like redundant.
In the current case I think that working with parameters would be realized later, and for now the code has only rudiments for future expanding, only logic realization. And it would be useful to work on code improvements when I will start to work with user parameters and uploading them to the application.

Reply:
Discussion moved to the issue https://github.com/alpheios-project/alignment-editor-new/issues/32